packages/forms/src/elements/file-list/components/File.tsx

```diff
@@ -25,32 +34,57 @@ export interface IFileProps extends HTMLAttributes<HTMLDivElement> {
   type?: FILE_TYPE;
   /** Applies inset `box-shadow` styling on focus */
   focusInset?: boolean;
+  /** Applies validation state styling */
+  validation?: VALIDATION_TYPE;
 }
 
 interface IStaticFileExport<T, P>
   extends ForwardRefExoticComponent<PropsWithoutRef<P> & RefAttributes<T>> {
   Close: typeof Close;
+  Delete: typeof Delete;
 }
 
 /* eslint-disable react/display-name */
 /**
  * @extends HTMLAttributes<HTMLDivElement>
  */
 export const File = forwardRef<HTMLDivElement, IFileProps>(
-  ({ children, type, focusInset, ...props }, ref) => (
-    <StyledFile {...props} focusInset={focusInset} ref={ref}>
-      {type && <StyledFileIcon>{fileIcons[type]}</StyledFileIcon>}
-      {Children.map(children, child => (typeof child === 'string' ? <span>{child}</span> : child))}
-    </StyledFile>
-  )
+  ({ children, type, isCompact, focusInset, validation, ...props }, ref) => {
+    const fileContextValue = useMemo(() => ({ isCompact }), [isCompact]);
+    const validationType = validation || type;
+
+    return (
+      <FileContext.Provider value={fileContextValue}>
+        <StyledFile
+          {...props}
+          isCompact={isCompact}
+          focusInset={focusInset}
+          validation={validation}
+          ref={ref}
+        >
+          {validationType && (
+            <StyledFileIcon isCompact={isCompact}>
+              {isCompact ? fileIconsCompact[validationType] : fileIconsDefault[validationType]}
+            </StyledFileIcon>
+          )}
+          {Children.map(children, child =>
+            typeof child === 'string' ? <span>{child}</span> : child
+          )}
```

Comment:
Yeah, if they are actual file names they would be unique. 

I believe it's expecting multiple components since there's the file's name, icon (Delete or Close), and the progress bar.

Reply:
We are only expecting one child. Either the consumer is going to give us a string, in which case it needs to be wrapped in order to contain the styling for overflow. Or they might elect to use Garden's [Ellipsis](https://garden.zendesk.com/components/typography#ellipsis-text) component, for example.